src/ucp/wireup/select.c

```diff
@@ -201,6 +202,11 @@ ucp_wireup_select_transport(ucp_ep_h ep, const ucp_address_entry_t *address_list
         iface_attr   = &worker->ifaces[rsc_index].attr;
         md_attr      = &context->tl_mds[context->tl_rscs[rsc_index].md_index].attr;
 
+        if ((context->tl_rscs[rsc_index].flags & UCP_CONTEXT_TLS_FLAG_AUX) &&
+            strcmp(criteria->title, UCP_WIREUP_AUX_CRITERIA_TITLE)) {
```

Comment:
maybe better to add rsc_flags to the criteria, instead of comparing the criteria title?
```
context->tl_rscs[rsc_index].allowed_rsc_flags = UCP_CONTEXT_TLS_FLAG_AUX
...
if (!ucs_test_all_flags(context->tl_rscs[rsc_index].flags, criteria->allowed_rsc_flags) { 
   continue;
}
```